samples/ask-akka-agent/src/main/java/akka/ask/agent/application/Knowledge.java

```diff
@@ -0,0 +1,42 @@
+package akka.ask.agent.application;
+
+import akka.ask.common.MongoDbUtils;
+import akka.ask.common.OpenAiUtils;
+import com.mongodb.client.MongoClient;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.rag.AugmentationRequest;
+import dev.langchain4j.rag.DefaultRetrievalAugmentor;
+import dev.langchain4j.rag.RetrievalAugmentor;
+import dev.langchain4j.rag.content.injector.ContentInjector;
+import dev.langchain4j.rag.content.injector.DefaultContentInjector;
+import dev.langchain4j.rag.content.retriever.EmbeddingStoreContentRetriever;
+import dev.langchain4j.rag.query.Metadata;
+
+public class Knowledge {
+  private final RetrievalAugmentor retrievalAugmentor;
+  private final ContentInjector contentInjector = new DefaultContentInjector();
+
+  public Knowledge(MongoClient mongoClient) {
+    var contentRetriever = EmbeddingStoreContentRetriever.builder()
+        .embeddingStore(MongoDbUtils.embeddingStore(mongoClient))
+        .embeddingModel(OpenAiUtils.embeddingModel())
+        .maxResults(10)
+        .minScore(0.1)
+        .build();
+
+    this.retrievalAugmentor = DefaultRetrievalAugmentor.builder()
+        .contentRetriever(contentRetriever)
+        .build();
+  }
+
+  public String addKnowledge(String question) {
+    var chatMessage = new UserMessage(question);
+    var metadata = Metadata.from(chatMessage, null, null);
```

Comment:
The metadata can also include `String chatMemoryId` and `List<ChatMessage> chatMemory`, but I don't think that is important, and probably not used by the DefaultRetrievalAugmentor.